src/components/ChannelList/ChannelList.js

```diff
@@ -1,743 +1,270 @@
-      filters,
-      sort,
-      options: queryOptions,
-    };
-  };
-
-  setRefreshingUIState = () =>
-    this.setStateAsync({
-      refreshing: true,
-      loadingChannels: false,
-      loadingNextPage: false,
-    });
-
-  setReloadingUIState = () =>
-    this.setStateAsync({
-      refreshing: false,
-      loadingChannels: true,
-      loadingNextPage: false,
-      error: false,
-      channels: Immutable([]),
-      channelIds: Immutable([]),
-    });
-
-  setLoadingNextPageUIState = () =>
-    this.setStateAsync({
-      refreshing: false,
-      loadingChannels: false,
-      loadingNextPage: true,
-    });
-
-  // Sets the loading UI state before the star
-  startQueryLoadingUIState = async (queryType) => {
-    switch (queryType) {
-      case 'refresh':
-        await this.setRefreshingUIState();
-        break;
-      case 'reload':
-        await this.setReloadingUIState();
-        break;
-      default:
-        await this.setLoadingNextPageUIState();
-        break;
-    }
-  };
-
-  finishQueryLoadingUIState = () =>
-    this.setStateAsync({
-      refreshing: false,
-      loadingChannels: false,
-      loadingNextPage: false,
-    });
+const ChannelList = (props) => {
+  const {
+    List = ChannelListMessenger,
+    filters = {},
+    lockChannelOrder = false,
+    onAddedToChannel,
+    onChannelDeleted,
+    onChannelHidden,
+    onChannelTruncated,
+    onChannelUpdated,
+    onMessageNew,
+    onRemovedFromChannel,
+    onSelect,
+    options = {},
+    setFlatListRef,
+    sort = {},
+  } = props;
+
+  const { client } = useContext(ChatContext);
+  const listRef = useRef(null);
+  const [forceUpdate, setForceUpdate] = useState(0);
+
+  const {
+    channels,
+    hasNextPage,
+    loadNextPage,
+    refreshList,
+    reloadList,
+    setChannels,
+    status,
+  } = usePaginatedChannels({
+    client,
+    filters,
+    options,
+    sort,
+  });
+
+  // Setup event listeners
+  useAddedToChannelNotification({ onAddedToChannel, setChannels });
+  useChannelDeleted({ onChannelDeleted, setChannels });
+  useChannelHidden({ onChannelHidden, setChannels });
+  useChannelTruncated({ onChannelTruncated, setChannels, setForceUpdate });
+  useChannelUpdated({ onChannelUpdated, setChannels });
+  useConnectionRecovered({ setForceUpdate });
+  useNewMessage({ lockChannelOrder, setChannels });
+  useNewMessageNotification({ onMessageNew, setChannels });
+  useRemovedFromChannelNotification({ onRemovedFromChannel, setChannels });
+  useUserPresence({ setChannels });
+
+  return (
+    <List
+      {...props}
+      channels={channels}
+      error={status.error}
+      forceUpdate={forceUpdate}
+      hasNextPage={hasNextPage}
+      loadingChannels={status.loadingChannels}
+      loadNextPage={loadNextPage}
+      loadingNextPage={status.loadingNextPage}
+      refreshing={status.refreshing}
+      refreshList={refreshList}
+      reloadList={reloadList}
+      setActiveChannel={onSelect}
+      setFlatListRef={(ref) => {
```

Comment:
@vishalnarkhede can you check that this ref logic is correct? i'm not totally sure